### PR TITLE
Origin/powerd

### PR DESCRIPTION
**Pull request by semafor**
Add a simple powerd mock that changes it's sysState and does wakeups.

### CODE REVIEW

File: dbusmock/templates/powerd.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+
+@dbus.service.method(MAIN_IFACE, in_signature='s', out_signature='')
+def disableProximityHandling(self, name):
+    pass
+
+
+@dbus.service.method(MAIN_IFACE, in_signature='s', out_signature='')
+def clearWakeup(self, cookie):
+    del self._wakeup_requests[cookie]
```
Comment (semafor) [inline dbusmock/templates/powerd.py]: Need to kill the timer as well, here.

File: dbusmock/templates/powerd.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        name, cookie, time
+    ))
+    self._wakeup_requests[cookie] = (name, time)
+    t = threading.Timer(
+        time - int(timelib.time()), wakeup, args=(name, cookie, time)
+    )
+    try:
+        t.start()
+    except (KeyboardInterrupt, SystemExit):
+        print("Cancelling foo")
```
Comment (semafor) [inline dbusmock/templates/powerd.py]: Will remove.

**Comment by martinpitt** [comment]
Apparently this is still work in progress; but TBH powerd is very specific to Ubuntu Touch, so I'm not sure if it's a good idea to ship this upstream. How many projects are going to need this?

File: dbusmock/templates/powerd.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        t.cancel()
+
+    return cookie
+
+
+@dbus.service.method(MAIN_IFACE, in_signature='s', out_signature='')
+def enableProximityHandling(self, name):
+    pass
+
```
Comment (martinpitt) [inline dbusmock/templates/powerd.py]: Please don't fully spell out these methods if they don't do anything, or only do something trivial. Please rather use `AddMethods()`  and add them wholesale, like the other templates do. This should get rid of most explicitly written methods here.

**Comment by semafor** [comment]
I didn't know powerd was specific to Ubuntu Touch, but if it is,  you're right. Closing for now.

**Pull request closed without merging**